src/transformState.js

```diff
@@ -5,7 +5,33 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  const resultArray = [];
+
+  for (let i = 0; i < actions.length; i++) {
+    switch (true) {
+      case actions[i].type === 'addProperties':
+        Object.assign(state, actions[i].extraData);
```

Comment:
don't mutate input state, add some constant

Reply:
- `state` is an object. You are supposed to add, change, or delete its
  properties instead of creating a new object

Reply:
I think I have to mutate state, don't I?